Jenkinsfile

```diff
@@ -150,9 +150,10 @@ pipeline {
 									'comma_ver='+swtVersions['comma_ver'], "new_comma_ver=${swtVersions['maj_ver']},${swtVersions['min_ver']},${swtVersions['new_rev']},0" ]) {
 								sh '''
 									# Delete native binaries to be replaced by subsequent binaries build
-									rm -f binaries/org.eclipse.swt.gtk.*/lib*-${swt_version}.so
-									rm -f binaries/org.eclipse.swt.win32.*/*-${swt_version}.dll
-									rm -f binaries/org.eclipse.swt.cocoa.*/lib*-${swt_version}.jnilib
+									swtNativeLibName='swt-*-+([0-9])r+([0-9])'
```

Comment:
Not sure what the `+`is doing, looks a wrong position in `swt-*-+([0-9])r+([0-9])` and what regex dialect is it?

Should it be `swt-.*-([0-9])+r([0-9])+`?

- swt 
- followed by dash 
- followed by anything
- followed by dash
- followed by one or more digits
- followed by "r"
- followed by one or more digits

Reply:
> Not sure what the `+`is doing, looks a wrong position in `swt-*-+([0-9])r+([0-9])` and what regex dialect is it?

I did it according to this document: https://www.gnu.org/software/bash/manual/html_node/Pattern-Matching.html

Couldn't find a specific dialect name but it looked official to me (I could mistaken).

> 
> Should it be ?
> 
>     * swt
> 
>     * followed by dash
> 
>     * followed by anything
> 
>     * followed by dash
> 
>     * followed by one or more digits
> 
>     * followed by "r"
> 
>     * followed by one or more digits

Yes that is the intended pattern.

Reply:
So this is working in bash 
```
org.eclipse.swt.gtk.linux.x86_64$ ll | grep -E "swt-.*-([0-9])+r([0-9])+"
-rwxrwxr-x  1 aloskuto users  39496 Feb 19 08:04 libswt-atk-gtk-4964r8.so
-rwxrwxr-x  1 aloskuto users  39496 Mar  6 08:12 libswt-atk-gtk-4965r2.so
-rwxrwxr-x  1 aloskuto users   6400 Feb 19 08:04 libswt-awt-gtk-4964r8.so
-rwxrwxr-x  1 aloskuto users   6400 Mar  6 08:12 libswt-awt-gtk-4965r2.so
-rwxrwxr-x  1 aloskuto users  44184 Feb 19 08:04 libswt-cairo-gtk-4964r8.so
-rwxrwxr-x  1 aloskuto users  44184 Mar  6 08:12 libswt-cairo-gtk-4965r2.so
-rwxrwxr-x  1 aloskuto users  10584 Feb 19 08:04 libswt-glx-gtk-4964r8.so
-rwxrwxr-x  1 aloskuto users  10584 Mar  6 08:12 libswt-glx-gtk-4965r2.so
-rwxrwxr-x  1 aloskuto users 602544 Feb 19 08:04 libswt-gtk-4964r8.so
-rwxrwxr-x  1 aloskuto users 602544 Mar  6 08:12 libswt-gtk-4965r2.so
-rwxrwxr-x  1 aloskuto users 454704 Feb 19 08:04 libswt-pi3-gtk-4964r8.so
-rwxrwxr-x  1 aloskuto users 454704 Mar  6 08:12 libswt-pi3-gtk-4965r2.so
-rwxrwxr-x  1 aloskuto users  51480 Feb 19 08:04 libswt-webkit-gtk-4964r8.so
-rwxrwxr-x  1 aloskuto users  51480 Mar  6 08:12 libswt-webkit-gtk-4965r2.so
```

```
org.eclipse.swt.win32.win32.x86_64$ ll | grep -E "swt-.*-([0-9])+r([0-9])+"
-rwxrwxr-x  1 aloskuto users 103816 Feb 19 08:04 swt-awt-win32-4964r8.dll
-rwxrwxr-x  1 aloskuto users 103816 Mar  6 08:12 swt-awt-win32-4965r2.dll
-rwxrwxr-x  1 aloskuto users 153992 Feb 19 08:04 swt-gdip-win32-4964r8.dll
-rwxrwxr-x  1 aloskuto users 153992 Mar  6 08:12 swt-gdip-win32-4965r2.dll
-rwxrwxr-x  1 aloskuto users 107400 Feb 19 08:04 swt-wgl-win32-4964r8.dll
-rwxrwxr-x  1 aloskuto users 107400 Mar  6 08:12 swt-wgl-win32-4965r2.dll
-rwxrwxr-x  1 aloskuto users 771464 Feb 19 08:04 swt-win32-4964r8.dll
-rwxrwxr-x  1 aloskuto users 771464 Mar  6 08:12 swt-win32-4965r2.dll
```

But the groups aren't needed, so this works too:
` ll | grep -E "swt-.*-[0-9]+r[0-9]+"`

Reply:
Looking at the build-logs I have the impression that it doesn't work on Jenkins, but to verify it I have replayed the Pipeline and adjusted it to push the resulting commit to the branch `check-PR-1100` in this repo.

Reply:
```
org.eclipse.swt.gtk.linux.x86_64$ rm swt-*-+([0-9])r+([0-9]).so
rm: cannot remove ‘swt-*-+([0-9])r+([0-9]).so’: No such file or directory
```

```
org.eclipse.swt.gtk.linux.x86_64$ ls | grep -E "swt-.*-[0-9]+r[0-9]+" | xargs echo
libswt-atk-gtk-4964r8.so libswt-atk-gtk-4965r2.so libswt-awt-gtk-4964r8.so libswt-awt-gtk-4965r2.so libswt-cairo-gtk-4964r8.so libswt-cairo-gtk-4965r2.so libswt-glx-gtk-4964r8.so libswt-glx-gtk-4965r2.so libswt-gtk-4964r8.so libswt-gtk-4965r2.so libswt-pi3-gtk-4964r8.so libswt-pi3-gtk-4965r2.so libswt-webkit-gtk-4964r8.so libswt-webkit-gtk-4965r2.so
```

```
org.eclipse.swt.win32.win32.x86_64$ ls | grep -E "swt-.*-[0-9]+r[0-9]+" | xargs echo
swt-awt-win32-4964r8.dll swt-awt-win32-4965r2.dll swt-gdip-win32-4964r8.dll swt-gdip-win32-4965r2.dll swt-wgl-win32-4964r8.dll swt-wgl-win32-4965r2.dll swt-win32-4964r8.dll swt-win32-4965r2.dll
```

So the command would be `ls | grep -E "swt-.*-[0-9]+r[0-9]+" | xargs rm`

Reply:
BTW, `binaries/org.eclipse.swt.gtk.*/lib` doesn't exist, there is no `lib` directory.

Reply:
So in bash this below works, should I push that?

```
swtNativeLibName='swt-.*-[0-9]+r[0-9]+'
ls -d binaries/org.eclipse.swt.gtk.*/*   | grep -E ${swtNativeLibName}.so     | xargs rm -vf
ls -d binaries/org.eclipse.swt.win32.*/* | grep -E ${swtNativeLibName}.dll    | xargs rm -vf
ls -d binaries/org.eclipse.swt.cocoa.*/* | grep -E ${swtNativeLibName}.jnilib | xargs rm -vf
```